sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -11,12 +11,45 @@ internal class CommunicationIdentifierSerializer
     {
         public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
         {
-           string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));
+            string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));
 
             AssertMaximumOneNestedModel(identifier);
 
-            if (identifier.CommunicationUser is CommunicationUserIdentifierModel user)
-                return new CommunicationUserIdentifier(AssertNotNull(user.Id, nameof(user.Id), nameof(CommunicationUserIdentifierModel)));
+            if (identifier.Kind is not null)
+            {
+                if (identifier.Kind == CommunicationIdentifierModelKind.CommunicationUser
+                    && identifier.CommunicationUser is not null)
```

Comment:
I wonder if we should support kind being set and the related object not being set...but I suppose if we don't break and fall back to the unknownidentifier and make no further assumptions about it, it's fine...
hmm, yeah, I suppose it's fine.